src/app/_components/NewsletterForm.tsx

```diff
@@ -19,16 +21,41 @@ export const NewsletterSchema = z.object({
 
 export type FormType = z.infer<typeof NewsletterSchema>
 
+const NEWSLETTER_URL = `${process.env.NEXT_PUBLIC_NEWSLETTER_SUBSCRIPTION_API_URL}/publications/${process.env.NEXT_PUBLIC_NEWSLETTER_SUBSCRIPTION_PUBLICATION_ID}/subscriptions`
+const AUTHORIZATION_HEADER = `Bearer ${process.env.NEXT_PUBLIC_NEWSLETTER_SUBSCRIPTION_API_KEY}`
+
+function postSubscription(email: string): Promise<number> {
+  return fetch(NEWSLETTER_URL, {
+    method: 'POST',
+    headers: {
+      Authorization: AUTHORIZATION_HEADER,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email }),
+  }).then((response) => response.status)
+}
```

Comment:
I extracted `postSubscription` to help separate concerns and keep the component logic clean.